# HG changeset patch
# User Raniere Silva <r.gaia.cs@gmail.com>
# Date 1357323327 7200
# Node ID 1f7b4c617a26e62fe30a1b33b0e1ef49db6f729b
# Parent  e8cd06038701f87074476ff75a110bebd5749b68
Version 0.1 of improvement of rcm().

* Improve the implementation of the Reverse Cuthill-McKee reordering.

Improve the function is important because it take effect in the performace of some decompositions.

Work supported by CNPq, Conselho Nacional de Desenvolvimento Cientifico e Tecnologico - Brasil.

diff --git a/sage/matrix/matrix2.pyx b/sage/matrix/matrix2.pyx
--- a/sage/matrix/matrix2.pyx
+++ b/sage/matrix/matrix2.pyx
@@ -13548,6 +13548,8 @@
                 r = x
                 L = Lx
             else:
+                if len(Lx) == len(L):
+                    x = self._min_degree(Lx[-1])
                 stop = True
         return x
 
@@ -13936,7 +13938,7 @@
             for j in xrange(self.ncols()):
                 if i != j and self[i, j]:
                     aux += 1
-            if aux < g:
+            if aux > g:
                 g = aux
                 r = i
         return r
